### ISSUE TITLE
The SDF library is old and not updated A replacement might be needed

### ISSUE THREAD

**Issue by EmergencyTemporalShift** [nortikin/sverchok#5421]
https://github.com/fogleman/sdf/tree/main has not been updated in two years and i believe it's package name has been replaced by another library: See https://pypi.org/project/SDF/ . Would https://github.com/peterropac/Aegolius be a suitable replacement?

**Comment by zeffii** [comment]
i tried `spomso` , it's unclear to me how you'd use it :)   , this is the lampshape 3d example converted to a SNLite script

```py
"""
in Size s
out Verts v
out Faces s
"""
# from sverchok.utils.modules import sdf_utils

import spomso
import numpy as np

from time import process_time

from spomso.cores.helper_functions import generate_grid, smarter_reshape
from spomso.cores.post_processing import hard_binarization
from spomso.cores.geom_2d import Arc

# ----------------------------------------------------------------------------------------------------------------------
# PARAMETERS

# size of the volume
co_size = 2.2, 2.2, 1.2
# resolution of the volume
co_resolution = 60, 60, 60

show = 'BINARY' #, FIELD
show_midplane = False
show_3d = True

# ----------------------------------------------------------------------------------------------------------------------
# COORDINATE SYSTEM
coor, co_res_new = generate_grid(co_size, co_resolution)

# ----------------------------------------------------------------------------------------------------------------------
# CREATE SDFs

start_time = process_time()

# define the cross-section of the lamp shade as an arc between angles PI and 1.35PI with a radius  of 1
shade = Arc(1, np.pi, np.pi + 0.7*np.pi/2)
# revolve the cross-section around an axis which is angled by PI/10 with respect to the y-axis.
# the radius of the revolution is 1 (radius of the arc) + 0.2
shade.axis_revolution(1 + 0.2, -np.pi/10)

# set the thickness of the lamp shade to 0.02
shade.rounding(0.02)

# position the lamp shade in space
shade.rotate(np.pi/10, (0, 0, 1))
shade.rotate(np.pi/2, (1, 0, 0))
shade.move((0, 0, 0.3))

# evaluate the SDF of the shade to create a signed distance field 3D map
shade_pattern = shade.create(coor)

end_time = process_time()
print("Evaluation Completed in {:.2f} seconds".format(end_time-start_time))

# ----------------------------------------------------------------------------------------------------------------------
# BINARIZATION
# distance field to a binary voxel map, where 1 corresponds to the interior and 0 to the exterior of the geometry.

if show=="BINARY":
    pattern = hard_binarization(shade_pattern, 0)


# PLOT

print("Drawing results...")
print(coor.shape)
print(pattern.shape)
print(pattern)
mask = pattern.astype(bool)
filtered = coor[:, mask]
Verts.append(filtered.T)
# Faces.append(faces)
```

<img width="982" height="547" alt="Image" src="https://github.com/user-attachments/assets/c429bd8a-831c-454b-9b37-577e4d44dcfa" />

**Comment by portnov** [comment]
@zeffii ,
"Marching cubes" node requires SvScalarField instance; that's a function that takes X/Y/Z and returns a number.
Can this library return such a function?

Or, alternatively, the algorithm which is used inside "marching cubes" node, wants a 3D numpy array, i.e. np.ndarray with shape (M, N, K), filled with numbers. Can this library return such array?

**Comment by EmergencyTemporalShift** [comment]
I'm not super attached to this library if better options exist, and I'll admit, I'm still learning about SDFs and marching cubes. I asked GitHub's ai (grain of salt I know) and it seems to think so. The output is as follows:

The peterropac/Aegolius library contains functionality that can work with scalar fields and produce the inputs required for a “marching cubes” node:

1. Function for Scalar Field (SDF):

• In the file `sdf_3D_jax.py`, functions like sdf_y, sdf_z (signed distance functions) take coordinates as input and return scalar values. This aligns closely with your requirement of creating an instance of SvScalarField.

2. Generating Numpy Arrays:

• The method `generate_grid` creates 3D grids of coordinates. These coordinates could potentially be evaluated with SDF or scalar field functions to produce a 3D numpy array, np.ndarray, filled with numbers.

3. Voxel Grids:

• The method `to_image` converts a point cloud to a voxel grid represented as a 3D numpy array.

These functionalities suggest that the library supports operations to generate scalar fields and 3D arrays necessary for marching cubes..."

**Comment by zeffii** [comment]
@portnov   with the hard binarization output (like quantization, sampling the surface/field at given intervals), this library will:
- output a (3, n) array , where n is the number of points in the field,
- output an additional array (a mask array) where a 0 is outside, and 1 is on the surface (or inside.. i'm not sure). 
    -  maybe it would be more useful to have ` distance is positive outside of the object, zero at its surface, and negative inside the object ` ( [link](https://pointersgonewild.com/2026-03-06-a-recursive-algorithm-to-render-signed-distance-fields/) )

Thats how i got this 
```py
# coor  is (3, n) array of coordinates
mask = pattern.astype(bool) # pattern is a mask list of length n, zeros and ones.
filtered = coor[:, mask]
Verts.append(filtered.T)
```
